src/serializer/functions.js

```diff
@@ -122,40 +122,69 @@ export class Functions {
     };
   }
 
-  checkReactRootComponents(statistics: ReactStatistics, react: ReactSerializerState): void {
+  _generateWriteEffectsForReactComponentTree(
+    componentType: ECMAScriptSourceFunctionValue,
+    effects: Effects,
+    simpleClassComponents: Set<Value>
+  ): void {
+    if (this.realm.react.output === "bytecode") {
+      throw new FatalError("TODO: implement React bytecode output format");
+    } else {
+      let additionalFunctionEffects = this._createAdditionalEffects(effects);
+      let value = effects[0];
+
+      if (value === this.realm.intrinsics.undefined) {
+        // if we get undefined, then this component tree failed and a message was already logged
+        // in the reconciler
+        return;
```

Comment:
If it failed, how does the bailout work? Where do we keep the original `createElement` call?

Reply:
The createElement/JSX isn't dealt with here. If we return, we are simply not flagging a function as being an `additionalFunction` at all.

Reply:
OK, that makes sense. Thanks.